m4a-dec.c

```diff
@@ -76,13 +75,11 @@ int main(int argc, char *argv[]) {
 	}
 
 	output_size = 8*2*2048;
-	output_buf = (uint8_t*) malloc(output_size);
-	decode_buf = (int16_t*) malloc(output_size);
+	decode_buf = (INT_PCM*) malloc(output_size/sizeof(INT_PCM));
```

Comment:
Why are you dividing the size (in bytes) by the size of INT_PCM here (resulting in a number of samples, not bytes) when you pass it to malloc (which wants a number of bytes, not samples)?

Reply:
If you want to make the size of INT_PCM configurable here, you should instead replace the `2` in `8*2*2048` by `sizeof(INT_PCM)`, as that's (maximum channels)*(size of sample)*(maximum frame).

Reply:
Ooops, typo

Reply:
You can mannually change INT_PCM to LONG and SAMPLE_BITS to 32 in AAC-FDK API header. In result AAC-FDK would compiles/works well.